packages/react-sdk/src/hooks/useAttachment.ts

```diff
@@ -15,77 +15,125 @@ import {
   getAttachmentData,
   updateAttachmentData,
 } from "@/helpers/caching/contentTypes/attachment";
+import { useMessage } from "@/hooks/useMessage";
+
+export type UseAttachmentOptions = {
+  autoload?: boolean;
+  autoloadMaxFileSize?: number;
+};
+
+export type AttachmentStatus =
+  | "init"
+  | "loading"
+  | "error"
+  | "loaded"
+  | "fileSizeLimitExceeded";
+
+// 10MB
+const defaultAutoloadMaxFileSize = 10 * 1024 * 1024;
 
 /**
  * This hook returns the attachment data of a cached message
  */
-export const useAttachment = (message: CachedMessage) => {
+export const useAttachment = (
+  message: CachedMessage,
+  options?: UseAttachmentOptions,
+) => {
   const { client } = useClient();
   const { db } = useDb();
-  const [isLoading, setIsLoading] = useState(false);
+  const { updateMessage } = useMessage();
   const [error, setError] = useState<Error | undefined>(undefined);
+  const [status, setStatus] = useState<AttachmentStatus>("init");
   const [attachment, setAttachment] = useState<Attachment | undefined>(
     undefined,
   );
 
-  const loadRemoteAttachment = useCallback(async () => {
-    if (
-      client &&
-      message.contentType === ContentTypeRemoteAttachment.toString()
-    ) {
-      // check if attachment data is already cached
-      const attachmentData = getAttachmentData(message);
-      if (attachmentData) {
-        setAttachment(attachmentData);
-        return;
-      }
+  const { autoload = true, autoloadMaxFileSize = defaultAutoloadMaxFileSize } =
```

Comment:
We'll need to make very clear in the documentation (and the type docs) that autoload is defaulted to `true` here. It's an optional field, so it's not obvious that leaving it undefined will actually turn it on. 

Alternatively, we could invert the flag and have something like `disableAutoload` that defaults to false.

Reply:
great point! i'll update the option to `disableAutoload`.

Reply:
[Address PR feedback](https://github.com/xmtp/xmtp-web/pull/72/commits/d58392438f89365b49f6fa68093fc44b5566bcd3)